xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -9237,4 +9237,17 @@ private XWikiAttachmentStoreInterface getXWikiAttachmentStoreInterface(String st
 
         return null;
     }
+
+    /**
+     * Compute and return the maximum authorized length for the full name (i.e. the serialized reference of the
+     * document) based on the current store limitation.
+     *
+     * @return the maximum authorized length for a document full name.
+     * @since 11.3RC1
+     */
+    @Unstable
+    public int getNameMaxLength()
```

Comment:
I don't think this is the right method name since what it return is the **full name** max length, the document **name** is only the last part of it.

Reply:
`getFullnameMaxLength` or do you have another suggestion?